buildSrc/src/main/kotlin/DownloadBinariesTask.kt

```diff
@@ -0,0 +1,58 @@
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import utils.createDirectoryIfNotExist
+import utils.createSymbolicLink
+import utils.download
+import utils.hasAllFiles
+import utils.isWindows
+import utils.unzipFile
+import java.nio.file.Files
+import java.nio.file.Paths
+
+open class DownloadBinariesTask : DefaultTask() {
+    private val projectRootDir = project.rootDir.toString()
+    private val outputDirectory = Paths.get(projectRootDir, "binaries")
+    private val resourceDirectory = Paths.get(projectRootDir, "test_runner", "src", "main", "resources", "binaries")
+
+    @Input
+    var forceUpdate: Boolean = false
+
+    @TaskAction
+    fun downloadBinaries() {
+        osName
+            .takeIf { forceUpdate || shouldDownloadBinaries() }
+            ?.let(::downloadAndUnzip)
+    }
+
+    private val osName: String
+        get() = if(isWindows) "windows" else "linux"
```

Comment:
Maybe add fail if mac?

Reply:
Basically, it should contain these binaries for running flank.jar on Linux while building on ubuntu

Reply:
Actually, I figure out that we should also have binaries for windows included. I will make logic to contains download both Windows and Linux when running the build on CI